### DISCUSSION TITLE
Static folder for user images

### DISCUSSION THREAD

**Discussion by ibragimof** [razonyang/hugo-theme-bootstrap#157] [Q&A] [answered]
### User images path

It requires to write path to user profile image & favicons from "Theme" folder. In this case if I will change static images, it overwrites by git submodule's next  update. How can I change user images path from "`/themes/hugo-theme-bootstrap/images/`" to "`/static/`', so don't overwrite images when updating theme?

**Comment by razonyang** [comment]
I assume that you mean the `/themes/hugo-theme-bootstrap/static/images/`.
Just create a fold called `static` in your project root, such as `/static`, and then put your images into the created directory.

Hugo will look up the images from your root `static` folder first, and then look up from the theme. For example, `/static/images/profile.webp` will override the image `/themes/hugo-theme-bootstrap/static/images/profile.webp`.

**Reply by ibragimof** [reply]
There is already root `/static` folder, generated by Hugo while initializing new site. But  seems config file "`\config\_default\author.toml`" looks only into theme's `/static/` folder.. I tried it not works for me.

_Second question:_ even after deleting "`profile.webp`" from "/themes/hugo-theme-bootstrap/static/images/profile.webp" -- wyh old (deleted) picture still exists on locally generated website (and why there is no errors after that in my console) ?

**Reply by razonyang** [reply]
It works nice for me, I tested it locally and remotely. 
See razonyang/hugo-theme-bootstrap-demo@462888c8a7d43a7899644da36ee71cf11a3ae8fc and preview it in https://hbs-demo.netlify.app/

**Comment by ibragimof** [comment, marked as answer]
So I think there is a bug issue.
Config file about changing **profile photo** (`\config\_default\author.toml`) **not works at all** with avatar pictures.
There is a markup file located in "**\themes\hugo-theme-bootstrap\layouts\partials\sidebar\profile\avatar.htm**l"  - its only way to change avatar photo and its path.

**Reply by razonyang** [reply]
Please double-check the avatar path. You can change the avatar path as `images/profile.png`, and then save your avatar image as `/static/images/profile.png`, if your avatar file is not placed in the right place, you'll get a 404 error in your console(F12 in Chrome).

**Reply by razonyang** [reply]
@ibragimof you're right, it's a bug, fixed by f08b5f983173fcbcce911339125563b815b07887, sorry about that.

**Comment by jehoshua7** [comment]
There are two "static" paths

`~/Hugo/mysite/static/images/`

`~/Hugo/mysite/themes/hugo-theme-bootstrap/exampleSite/static/images/`

If I want posts (content) to display an image, where should the image reside ?

**Comment by jehoshua7** [comment]
After some testing - when the image was in ~/Hugo/mysite/static/images/

    `ERROR 2021/09/18 18:58:28 no such image can be found in page resources and the static folder`
    
put the image in ~/Hugo/mysite/themes/hugo-theme-bootstrap/exampleSite/static/images/ , and **no error**
